src/client/Client.cc

```diff
@@ -9610,13 +9609,14 @@ int Client::_read_sync(Fh *f, uint64_t off, uint64_t len, bufferlist *bl,
 		      &onfinish);
     client_lock.unlock();
     int r = onfinish.wait();
-    client_lock.lock();
 
     // if we get ENOENT from OSD, assume 0 bytes returned
     if (r == -ENOENT)
       r = 0;
-    if (r < 0)
+    if (r < 0) {
+      client_lock.lock();
```

Comment:
Can we use a goto instead? This is really awkward and error prone.

Reply:
Yeah, sure, will fix it.